### ISSUE TITLE
Coverage report is generating for a springboot + maven project, but there is an NPE

### ISSUE THREAD

**Issue by Ruhshan** [Kotlin/kotlinx-kover#701]
Hello,

I have just added kover with my spring boot project. I can run tests and also generate coverage reports. But there is this NPE:

![img](https://github.com/user-attachments/assets/afe52bed-d76b-4723-827b-e12efde7c3eb)

Though the coverage is generated. Can you please tell me I'm doing something wrong? Or what could resolve this.

Here is a minimal project that generates this issue:

https://github.com/Ruhshan/springboot-maven-kt

**Comment by shanshin** [comment]
Hi,
thanks for the report and the reproducer!

To workaround this bug you may add empty `sourceDirs` element to the `kotlin-maven-plugin` configuration.
Like
```
            <plugin>
                <groupId>org.jetbrains.kotlin</groupId>
                <artifactId>kotlin-maven-plugin</artifactId>
                <configuration>
                    <args>
                        <arg>-Xjsr305=strict</arg>
                    </args>
                    <compilerPlugins>
                        <plugin>spring</plugin>
                    </compilerPlugins>
                    <sourceDirs></sourceDirs> <!-- fix Kover NPE -->
                </configuration>
                <dependencies>
                    <dependency>
                        <groupId>org.jetbrains.kotlin</groupId>
                        <artifactId>kotlin-maven-allopen</artifactId>
                        <version>${kotlin.version}</version>
                    </dependency>
                </dependencies>
            </plugin>
```

**Comment by shanshin** [comment]
Fixed in `0.9.0`
